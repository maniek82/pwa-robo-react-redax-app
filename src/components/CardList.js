import React from 'react';
import Card from './Card';


const CardList = ({users}) => {

    const cardComponent = users.map((user,i)=>{
            return (
            <Card 
               key={i}
              
              name={user.name}
              email={user.email}
              city={user.region}
              age={user.age}
              photo = {user.photo}  
              phone = {user.phone}  
            />
            )
        })
    return ( 
        <div>
           
            {cardComponent}
        </div>
     );
}
 
export default CardList;