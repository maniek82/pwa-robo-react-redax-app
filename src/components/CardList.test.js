import {shallow} from 'enzyme'
import React from 'react'
import CardList from './CardList';



describe('CardList snapshot', () => {


it('expect to render CardList component',()=> {
    const mockRobots = [
        {
            key:1,
            name: 'John',
            email: 'maniek@gmail.com',
            city:'leicester',
            age:5,
            photo: 'photo',  
            phone: 'phone' 
        }
    ]
    expect(shallow(<CardList users={mockRobots}/>)).toMatchSnapshot()

})

})
