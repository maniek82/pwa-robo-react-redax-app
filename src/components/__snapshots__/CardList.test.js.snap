// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList snapshot expect to render CardList component 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <Card
    age={5}
    email="maniek@gmail.com"
    key="0"
    name="John"
    phone="phone"
    photo="photo"
  />
</div>
`;
